mlflow/utils/search_utils.py

```diff
@@ -37,19 +36,41 @@ def filter_string(self):
     def search_expressions(self):
         return self._search_expressions
 
+    @classmethod
+    def _trim_ends(cls, string_value):
+        return string_value[1:-1]
+
+    @classmethod
+    def _is_quoted(cls, value, pattern):
+        if (value.startswith(pattern) and not value.endswith(pattern)) or \
+                (not value.startswith(pattern) and value.endswith(pattern)):
+            raise MlflowException("Mismatched quote types in argument {}" % value,
```

Comment:
Great catch! I cannot make up my mind whether `%..` or `{} + format` is a better way to stringyfying messages.

Reply:
The "Python" way of doing it is `.format()`. https://stackoverflow.com/a/12382738

Reply:
let's python!

Reply:
> The tests still seem to pass, which means that we aren't testing this code path. Can we add a relevant test?

Looks like this is caught up ahead -- because a mismatched quote is not valid syntax and `sqlparse` catches that. So yeah! Removing this code, 'cuz deleting code is so much better than writing test cases :)